src/gui/plugins/scene3d/Scene3D.cc

```diff
@@ -2207,6 +2208,19 @@ void Scene3D::Update(const UpdateInfo &_info,
 
     auto renderWindow = this->PluginItem()->findChild<RenderWindowItem *>();
     renderWindow->SetWorldName(this->dataPtr->worldName);
+    auto worldEntity =
+      _ecm.EntityByComponents(components::Name(this->dataPtr->worldName),
+        components::World());
+    auto renderEngineGuiComp =
+      _ecm.Component<components::RenderEngineGuiPlugin>(worldEntity);
+    if (renderEngineGuiComp && !renderEngineGuiComp->Data().empty())
+    {
+      this->dataPtr->renderUtil->SetEngineName(renderEngineGuiComp->Data());
+    }
+    else
+    {
+      ignwarn << "Render engine not found" << std::endl;
```

Comment:
This sounds a bit misleading. How about an `igndbg` with something along the lines of "RenderEngineGuiPlugin component not found, render engine won't be set from the ECM".

Reply:
88cd727c47443ca8fa46d52294199c4136ff7be4